### PR TITLE
[rsm-binary-io] Add package

### PR DESCRIPTION
**Pull request by Qudix**
Add `rsm-binary-io` crossplatform binary io library

### CODE REVIEW

**Comment by Qudix** [comment]
linux-ubuntu is failing due to the standard include `<span>` not being found. The library notes that `gcc10` is required likely due to this.

File: packages/r/rsm-binary-io/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    add_urls("https://github.com/Ryan-rsm-McKenzie/binary_io/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/Ryan-rsm-McKenzie/binary_io.git")
+    add_versions("2.0.5", "4cc904ef02f77e04756cbdf01372629b0f04d859f06ee088d854468abdd4b840")
+
+    on_install(function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            set_languages("c++20")
+            target("rsm-binary-io")
+                set_kind("static")
```
Comment (waruqi) [inline packages/r/rsm-binary-io/xmake.lua:15]: and shared

**Comment by Qudix** [comment]
What about gcc/g++?

**Comment by waruqi** [comment]
> What about gcc/g++?

? The gcc for the linux ci is rather old, I see that other archlinux ci's have passed.

**Pull request merged**